Core/Resources/Archives/Collection/ArchiveCollection.cs

```diff
@@ -94,6 +95,7 @@ public class ArchiveCollection : IResources, IPathResolver
     private string? m_lastLoadedMapPath;
     private bool m_lastLoadedMapIsTemp;
     private bool m_initTextureManager;
+    private readonly Dictionary<(string, ResourceNamespace), BrightmapDefinition?> m_brightmapLookupCache = [];
```

Comment:
I think we should make this a struct that implements IEquatable. I looked at the source it looks like it may go through a generic equals with object which unboxes and allocates so I would rather be safe since it's easy.

Also I think this may have a problem with the hashing and equals being case sensitive which we wouldn't want so that would need to be addressed in the IEquatable implementation.

https://github.com/Helion-Engine/Helion/blob/dev/Core/Geometry/Vectors/Vec2D.cs